src/core/qgsmessagelog.h

```diff
@@ -51,7 +51,8 @@ class CORE_EXPORT QgsMessageLog : public QObject
      * If it is FALSE, the message should appear in logs silently. Note that log viewer implementations may
      * only respect notification hints for certain message levels.
      */
-    static void logMessage( const QString &message, const QString &tag = QString(), Qgis::MessageLevel level = Qgis::MessageLevel::Warning, bool notifyUser = true );
+    static void logMessage( const QString &message, const QString &tag = QString(), Qgis::MessageLevel level = Qgis::MessageLevel::Warning, bool notifyUser = true,
+                            const char *file = __builtin_FILE(), const char *function = __builtin_FUNCTION(), int line = __builtin_LINE() );
```

Comment:
OK, so after our tests, please add this kind of TODO for C++20:

```suggestion
// TODO: Update this code to use std::source_location from C++20 when transitioning to a fully C++20-compliant codebase.
//       Currently, we rely on __builtin_XXX functions (e.g., __builtin_FILE(), __builtin_LINE()), 
//       which have been successfully tested across multiple systems (Windows, macOS, Linux, FreeBSD) 
//       and compilers (LLVM, GCC, MSVC).

// Note: We tested with LLVM on FreeBSD and macOS, and std::experimental::source_location is not available. 
//       It works fine with GNU. It also seems unavailable with MSVC.
//       For now, we stick with __builtin_XXX because it is "portable" and functional across all tested environments. 
//       We'll switch to std::source_location once the transition to C++20 is complete.

    static void logMessage( const QString &message, const QString &tag = QString(), Qgis::MessageLevel level = Qgis::MessageLevel::Warning, bool notifyUser = true,
                            const char *file = __builtin_FILE(), const char *function = __builtin_FUNCTION(), int line = __builtin_LINE() );
```